### PR TITLE
Handle public/private profiles

### PR DESCRIPTION
**Pull request by tsauerwein**
Closes https://github.com/c2corg/v6_api/issues/425

### CODE REVIEW

File: c2corg_api/views/user_profile.py [hunk 1]
```diff
@@ -22,9 +24,32 @@ def collection_get(self):
...
+            raise HTTPNotFound('user not found')
+        elif requested_user.is_profile_public or \
+                self.request.has_permission('authenticated'):
+            # only show the full profile if authenticated or if the user marked
+            # the profile as public
+            return self._get(UserProfile, schema_user_profile)
+        else:
+            # otherwise only return the user name
+            return {
+                'not_authenticated': True,
```
Comment (asaunier) [inline c2corg_api/views/user_profile.py]: What about returning ``not_authorized`` instead?

Comment (tsauerwein) [inline c2corg_api/views/user_profile.py]: To be nit-picky: Every user that is authenticated can see every profile, no special authorization is required. So, `not_authenticated` is more correct. ;)

Comment (asaunier) [inline c2corg_api/views/user_profile.py]: I think ``not_authorized`` is clearer: it's the main reason why you don't see the full profile. ``not authenticated`` is the partial reason why you're not authorized (the other part being that the profile is available to authenticated users only). By the way, in the future users might choose to make their profile not available to anyone => then every request would be ``not authorized`` (even for authenticated people).

**Comment by tsauerwein** [comment]
Ok, I renamed to `not_authorized`. Thanks for the review!

**Comment by asaunier** [comment]
Thanks for the change!

**Pull request merged**